### PR TITLE
feat(forms): hide semantic Radio, Checkbox, and Toggle inputs more inclusively

### PR DESCRIPTION
**Pull request by jgorfine-zendesk**
## Description

This PR aims to hide the inputs that are part of the Checkbox, Radio, and Toggle components in a more inclusive way. This technique comes from Sara Soueidan via her blog post, “Inclusively Hiding & Styling Checkboxes and Radio Buttons:” [Hiding the checkboxes inclusively](https://www.sarasoueidan.com/blog/inclusively-hiding-and-styling-checkboxes-and-radio-buttons/#hiding-the-checkboxes-inclusively).

## Detail

As Sara articulates in her blog post, there are assistive technologies — such as TalkBack, on Android phones — that enable low-vision and blind users to explore the web by touch. As far as checkboxes and radio buttons go, this means that the actual, semantic HTML inputs need to be the same size (i.e., occupy the same space) as the decorative inputs, so that someone exploring by touch can feel them and interact with them. 

### Before

The inputs were smaller than their decorative counterparts:

Before: less inclusive radio button

Before: less inclusive checkbox

Before: less inclusive toggle

### After (this PR)

The inputs are the same size as their decorative counterparts. They're positioned on top of the decorative inputs, and made completely transparent:

After: more inclusive radio button

After: more inclusive checkbox

After: more inclusive toggle

## Checklist

- [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [ ] :globe_with_meridians: demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [ ] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :guardsman: includes new unit tests
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

**Comment by jgorfine-zendesk** [comment]
[Update Garden checkbox and radio button styles to be more inclusive](https://trello.com/c/9wBWAdb9/38-update-garden-checkbox-and-radio-button-styles-to-be-more-inclusive)

**Comment by jzempel** [comment]
Thanks @jgorfine-zendesk. Please note that `Toggle` is part of this family as well. It would be great if you can branch internally vs. fork. Then we'll get your PR running through CI and SHA deployments which makes review more straightforward.

**Comment by jgorfine-zendesk** [comment]
Hi @jzempel — I'll be sure to take a look at the Toggle and include it in this effort. 🙂 

> It would be great if you can branch internally vs. fork
Would you mind taking another look? I could I have sworn I did it this way already, but maybe I'm missing something.

**Comment by jzempel** [comment]
> Would you mind taking another look? I could I have sworn I did it this way already, but maybe I'm missing something.

oic, yeah it looks good.

**Comment by Francois-Esquire** [comment]
@jgorfine-zendesk I like this approach from Sara Soueidan and appreciate the PR. I am a little confused, were we expecting to see `Checkbox` included as well? I think this is ready to review tho, would love to see this PR/branch deployed - maybe we can take it out of draft?

**Comment by jgorfine-zendesk** [comment]
> @jgorfine-zendesk I like this approach from Sara Soueidan and appreciate the PR. I am a little confused, were we expecting to see `Checkbox` included as well? I think this is ready to review tho, would love to see this PR/branch deployed - maybe we can take it out of draft?

Hi @Francois-Esquire — the work that's currently in the draft should cover both the Radio and Checkbook component; however, it does not include the Toggle, and @jzempel requested that it does. 

At this point, we could either:
1) Move forward with the Radio and Checkbox, and then I can come back to the Toggle sometime in the next couple of weeks
2) Keep waiting until the Toggle has also been addressed, and then merge in this PR. 

Would it be helpful if I left a follow up comment here articulating why I'm blocked (technically speaking) on the Toggle work?

**Comment by jzempel** [comment]
Couple clarifying points here...

1. Given the nature of inheritance between these `Radio`, `Checkbox`, and `Toggle` really need to be addressed as a unit of work
2. Draft PRs will deploy, but it looks like the build sees @jgorfine-zendesk as [Unauthorized](https://app.circleci.com/pipelines/github/zendeskgarden/react-components?branch=jgorfine%2Fstyled-checkboxes-and-radios-inclusively). Jen, please authenticate with that link. At that point, CircleCI should be aware of your write permissions and allow future deploys to go through

File: packages/forms/src/styled/radio/StyledRadioInput.ts
Comment (jzempel) [inline packages/forms/src/styled/radio/StyledRadioInput.ts:143]: Without a [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) reset, these visual artifacts are susceptible to being hidden by an intervening stacked layer (i.e. absolute `div`). I think there are a couple of things that could be done here:
- Add `z-index: 0;` to the `StyledRadioLabel`
-or-
- Update the `Radio`, `Checkbox`, `Toggle` element components to render the styled input _after_ the `{children}` and remove the `z-index: -1;` here

I'd generally be in favor of the latter over trying to manipulate stacking contexts – which always feels a bit messy.  But only if it can be done with minimal CSS updates. If we find ourselves updating a bunch of CSS to make it work, then I'd advise sticking with the former.

The problem can be seen with Bedrock enabled in Storybook: https://61e9d60fd1320011fdcba22c--zendeskgarden-react-components.netlify.app/?path=/story/packages-forms-fieldset--checkbox&globals=bedrock:enabled

File: packages/forms/src/styled/radio/StyledRadioInput.ts
Comment (jzempel) [inline packages/forms/src/styled/radio/StyledRadioInput.ts:153]: Ditto :wink:

**Review by jzempel — CHANGES_REQUESTED** [review]

File: packages/forms/src/styled/radio/StyledRadioInput.ts
Comment (jzempel) [reply packages/forms/src/styled/radio/StyledRadioInput.ts:143]: Actually, the more I think about it, I think there are a bunch of sibling selectors that break if the `{children}`/input are reordered. So I'd recommend sticking with the `z-index: 0;` added to `StyledRadioLabel`.

Comment (jgorfine-zendesk) [reply packages/forms/src/styled/radio/StyledRadioInput.ts]: @jzempel — thanks for showing me the Bedrock thing! (I didn't know about it before.)

> I'd generally be in favor of the latter over trying to manipulate stacking contexts – which always feels a bit messy.
I totally agree with you here. In this particular case, I think part of the challenge comes from the faux input (the `label::before`) being positioned absolutely, as well as the `input`. I can't help but wonder if styling whole component with Flexbox instead might help alleviate this. 🤔 

RE the `z-index`: I just tried it out again locally — both with and without Bedrock — and it looks like we might not need to hard-set the `z-index` after all? I just pushed up a commit that removes it altogether. Would you mind taking another look?

**Review by jzempel — APPROVED** [review]
@jgorfine-zendesk thanks for this contribution, representing an inclusive uplift to an often used class of Garden components. With one more @zendeskgarden/maintainers review, we'll get this merged and published ASAP.

**Review by Francois-Esquire — APPROVED** [review]
PR looks great! Thank you @jgorfine-zendesk 🙌

**Review by hzhu — APPROVED** [review]

**Comment by jgorfine-zendesk** [comment]
My pleasure, @jzempel! Thank you for the opportunity. 😄

**Pull request merged**
